pkg/config/mesh/meshwatcher/watcher_test_utils.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package meshwatcher
+
+import (
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/kube/krt"
+)
+
+type FixedWatcher struct {
+	adapter
+	col krt.StaticSingleton[MeshConfigResource]
+}
+
+func (w FixedWatcher) Set(n *meshconfig.MeshConfig) {
+	w.col.Set(&MeshConfigResource{n})
+}
+
+// NewFixedWatcher creates a new Watcher that always returns the given mesh config.
+func NewFixedWatcher(m *meshconfig.MeshConfig) FixedWatcher {
```

Comment:
Ah FixedWatcher ~= testwatcher because it's in test utils

Reply:
I've renamed this to make it more clear